x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/KnnDenseVector.java

```diff
@@ -99,15 +81,15 @@ public double l2Norm(QueryVector queryVector) {
     @Override
     public double cosineSimilarity(float[] queryVector, boolean normalizeQueryVector) {
         if (normalizeQueryVector) {
-            return dotProduct(queryVector, DenseVector.getMagnitude(queryVector)) / getMagnitude();
+            return (dotProduct(queryVector) / DenseVector.getMagnitude(queryVector)) / getMagnitude();
```

Comment:
Small comment, we usually would do `dotProduct(queryVector) / (DenseVector.getMagnitude(queryVector) * getMagnitude());` since it's more readable (similar to how you'd write it mathematically) and potentially a little faster.

Reply:
Makes sense, changed.